KafkaIngester/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -339,6 +342,88 @@ public static Table consumeToTable(
         return streamToTableAdapter.table();
     }
 
+    public static Table consumeJsonToTable(
+            @NotNull final Properties consumerProperties,
+            @NotNull final String topic,
+            @NotNull final IntPredicate partitionFilter,
+            @NotNull final IntToLongFunction partitionToInitialOffset,
+            final ColumnDefinition<?>[] valueColumns) {
+        return consumeJsonToTable(consumerProperties, topic, partitionFilter, partitionToInitialOffset, valueColumns, null);
+    }
+
+    public static Table consumeJsonToTable(
+            @NotNull final Properties consumerProperties,
+            @NotNull final String topic,
+            @NotNull final IntPredicate partitionFilter,
+            @NotNull final IntToLongFunction partitionToInitialOffset,
+            final ColumnDefinition<?>[] valueColumns,
+            final Map<String, String> columnNameToJsonField) {
+        final int nCols = 3 + valueColumns.length;
+        final ColumnDefinition<?>[] commonColumns = new ColumnDefinition[nCols];
+        int c = 0;
+        c += getCommonCols(commonColumns, 0, consumerProperties, partitionFilter == ALL_PARTITIONS);
+        final int[] commonColsIndices = new int[3];
```

Comment:
We are repeating this thing over and over; in the next iteration we should refactor this to be common.

Reply:
That is the idea, my next step is to refactor all the 3 calls into one, with arguments that are explicit avro versus json versus simple "specs".